torch/_inductor/runtime/benchmarking.py

```diff
@@ -200,6 +219,59 @@ def benchmark_many_gpu(
             self.benchmark_gpu(_callable, *args, **kwargs) for _callable in callables
         ]
 
+    @time_and_count
+    def lazy_benchmark(
+        self: Self,
+        fn: Callable[..., Any],
+        fn_args: Tuple[Any],
+        fn_kwargs: Dict[str, Any],
+        **kwargs: Any,
+    ) -> Union[LazyBenchmark, float]:
+        """Lazily enque `fn(*fn_args, *fn_kwargs)` for benchmarking. Follows the same
+        principles as `Benchmarking.benchmark` except that the benchmarking is delayed,
+        in the form of returning a `LazyBenchmark` object, such that benchmarking does
+        not occur until the benchmarking results are required. This allows us to have
+        better benchmark grouping (i.e. we can utilize `GroupedInductorBenchmarker`)
+        without having to refactor user callsites.
+        Arguments:
+        - fn: The function to benchmark.
+        - fn_args: The function's arguments.
+        - fn_kwargs: The function's kwargs.
+        Keyword Arguments:
+        - **kwargs: The benchmarking implementation's kwargs.
+        Returns:
+        - A `LazyBenchmark`, `x`, that evaluates to the actual benchmark result
+        when `float(x)` or `x.timing` is executed.
+        """
+        inferred_device = self.infer_device_type(fn_args, fn_kwargs)
+        _callable = lambda: fn(*fn_args, **fn_kwargs)  # noqa: E731
+        if inferred_device == torch.device("cpu"):
+            return self.lazy_benchmark_cpu(_callable, **kwargs)
+        return self.lazy_benchmark_gpu(_callable, **kwargs)
+
+    @time_and_count
+    def lazy_benchmark_cpu(
+        self: Self, _callable: Callable[[], Any], *args: Any, **kwargs: Any
+    ) -> LazyBenchmark:
+        # we need to execute `_callable` once before creating the lazy
+        # benchmark, since we want any exceptions that `_callable` might
+        # throw to be thrown when this function is called, not when the
+        # lazy benchmark is finally evaluated
+        _callable()
+        return LazyBenchmark(lambda: self.benchmark_cpu(_callable, *args, **kwargs))
+
+    @time_and_count
+    def lazy_benchmark_gpu(
+        self: Self, _callable: Callable[[], Any], *args: Any, **kwargs: Any
+    ) -> LazyBenchmark:
+        # we need to execute `_callable` once before creating the lazy
+        # benchmark, since we want any exceptions that `_callable` might
+        # throw to be thrown when this function is called, not when the
+        # lazy benchmark is finally evaluated
+        _callable()
+        torch.cuda.synchronize()
```

Comment:
Current benchmark_gpu support cuda/xpu, but here you hard code CUDA, that will break XPU, suggest to use common device interface : 
```
device_interface = get_interface_for_device(device_type)
device_interface.synchronize().
```